sn_networking/src/record_store.rs

```diff
@@ -428,27 +428,41 @@ impl NodeRecordStore {
     /// Calculate the cost to store data for our current store state
     #[allow(clippy::mutable_key_type)]
     pub(crate) fn store_cost(&self, key: &Key) -> (NanoTokens, QuotingMetrics) {
-        let quoting_metrics = QuotingMetrics {
-            records_stored: self.records.len(),
-            max_records: self.config.max_records,
-            received_payment_count: self.received_payment_count,
-            live_time: self.timestamp.elapsed().as_secs(),
-        };
+        let live_time = self.timestamp.elapsed().as_secs();
+        let mut records_stored = self.config.max_records;
 
         let cost = if self.contains(key) {
-            0
+            NanoTokens::zero()
+        } else if let Some(distance_range) = self.distance_range {
+            let record_keys_as_hashset: HashSet<&Key> = self.records.keys().collect();
+            let relevant_records =
+                self.get_records_within_distance_range(record_keys_as_hashset, distance_range);
```

Comment:
looks like the correspondent `addr_distribution_sim` test not got updated to reflect this change?
this will result in an un-sync between the sim and real code.

Reply:
Hmmm, im not sure what needs to be changed here. 

I've updated some naming, and made things a bit clearer, but the crux of the test is entirely the same. We were already only calculating for close_records as there's no churn here.